forge/db/controllers/Device.js

```diff
@@ -1,7 +1,45 @@
 const SemVer = require('semver')
 const { literal } = require('sequelize')
 
+/**
+ * inflightState - when devices are transitioning between states, there
+ * is no need to store that in the database. But we do need to know it so the
+ * information can be returned on the API.
+ */
+const inflightState = { }
```

Comment:
We cannot have inflight state - it will prevent us from being able to horizontally scale the app. Each API request could hit a different instance of the platform.

I haven't looked to see how exactly this state is used in this PR but will need addressing.

Reply:
hmmmm. good point and no simple solution. 

Will need to study.

Reply:
@knolleary I have removed the in-memory inflight state and reverted to updating the device.  tests updated to reflect this.

the UI already optimistically sets state so when the real status comes in from the API, it will align with the real DB value.

Operationally tested and works consistently.

This approach should be suitable for scaling.

Update incoming.

Reply:
Update here: [remove inflight device status](https://github.com/FlowFuse/flowfuse/pull/3548/commits/ae52707d70d7a1d71c1921ad65d6f4abfdf02dae)